src/tools/diagnostics.md

```diff
@@ -0,0 +1,2436 @@
+---
+title: Diagnostics
+description: Details for diagnostics produced by the Dart analyzer.
+---
+
+This page lists all diagnostic messages produced by the Dart analyzer,
+with details about what those messages mean
+and how you can fix your code.
+For more information about the analyzer, see
+[Customizing static analysis](/guides/language/analysis-options).
+
+
+## Glossary
```

Comment:
I have a slight preference for the top of the page, because some people walk up the URL hierarchy, and those that do are the thorough types who might actually read the glossary.

Reply:
Will do.